.github/workflows/ci.yaml

```diff
@@ -5,44 +5,52 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
+    timeout-minutes: 30
+    defaults:
+      run:
+        shell: bash -l {0}
     strategy:
       fail-fast: false
       matrix:
         jobqueue: ["htcondor", "pbs", "sge", "slurm", "none"]
 
     steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.7.0
+        with:
+            access_token: ${{ github.token }}
       - name: Checkout source
         uses: actions/checkout@v2
 
       - name: Setup Conda Environment
         uses: conda-incubator/setup-miniconda@v2
         with:
-          miniconda-version: "latest"
-          python-version: 3.6
-          environment-file: ci/environment.yml
+          channels: conda-forge
+          mamba-version: "*"
           activate-environment: dask-jobqueue
           auto-activate-base: false
+         
+      - name: Setup conda environment 
+        run: |
+          mamba env update -f ci/environment.yml
+          mamba list
```

Comment:
As I understand it, the current setup doesn't use this new environment. We end up using the environment created during the docker container creation:

https://github.com/dask/dask-jobqueue/blob/127470780ca82cfabecff247ee5e306867606a83/ci/slurm/Dockerfile#L9-L11

I'm wondering.. should we just include  schedulers(pbs, slurm, etc..) _only_ in the docker containers and then create the conda environment (during CI runs) and mount conda and its envs directories into the docker images during test runs? 

It's my understanding that this would:

- make it easy to make changes to the conda environment in a PR without having to mess with the docker container. 
- reduce the docker container size (as a bonus)... 

Any thoughts?

Reply:
Yep, if you can make that works, I think this would be a nice change to decouple Docker images (base infrastructure) and conda env (real dask-jobqueue use case).

Should we just merge this one and work on this later?

Reply:
Great! 

> Should we just merge this one and work on this later?

👍🏽 Yes, let's merge this as is. I will submit a separate PR to address the docker/conda decoupling.